flaml/automl/task/task.py

```diff
@@ -345,11 +342,11 @@ def estimator_class_from_str(cls, estimator_name: str) -> "flaml.automl.ml.BaseE
         Raises:
             ValueError: The provided estimator_name has not been registered for this task type.
         """
-        if estimator_name in cls.estimators:
-            return cls.estimators[estimator_name]
+        if estimator_name in self.estimators:
+            return self.estimators[estimator_name]
```

Comment:
It looks that `self.estimators` is indeed called often. The performance issue thus exists. Try to make the import happen only once by storing the dict in a private class member and don't compute it every time.

Reply:
Done

Reply:
```suggestion
        estimator = self.estimators.get(estimator_name)
        if estimator is not None:
            return estimator
```

Reply:
Why? I find my formulation more readable, and the difference in speed will be in the single-digit milliseconds; if this was called thousands of times, we might want to count milliseconds, but this is called once per estimator evaluation, so the speed improvement will be negligible.